wgpu-hal/src/gles/device.rs

```diff
@@ -403,11 +443,12 @@ impl super::Device {
             }
         }
 
-        Ok(super::PipelineInner {
+        Ok(Arc::new(super::PipelineInner {
             program,
             sampler_map,
             uniforms,
-        })
+            ref_count: AtomicU32::new(0),
```

Comment:
Why a separate refcount vs using the Arc's refcount?

Reply:
Good point!